controller/volume_controller.go

```diff
@@ -3198,6 +3198,10 @@ func (c *VolumeController) createEngine(v *longhorn.Volume, isNewEngine bool) (*
 
 	if isNewEngine {
 		engine.Spec.Active = true
+	} else {
+		if v.Spec.MigrationNodeID != "" {
+			engine.Spec.NodeID = v.Spec.MigrationNodeID
+		}
```

Comment:
Why need to do this? this has been handled in https://github.com/longhorn/longhorn-manager/blob/023dd3cbd5009f390608ea9cdd9f31bd397cf8a2/controller/volume_controller.go#L3888?

Reply:
Because I found that updating engine spec for the migrationEngine will be failed sometimes and it would never run this part of codes again for some reasons such as `v.Spec.MigrationNodeID` had been set "" or old engine had been deleted.
Therefore I set the engine nodeID as v.Spec.MigrationNodeID when creating the migrating engine.

Reply:
Wait... Is this engine the old one? Can we directly change the running old engine spec node ID?

Reply:
The migration engine is the new created one.

Reply:
Here you are operating `engine`

Reply:
The migrationEngine is created here for `isNewEngine` is `false`

Reply:
Removed this assignment, it will have been set again when synchronizing the snapshots if there is a conflict for updating the migration engine.
Thanks.